plugins/package-managers/node/src/main/kotlin/yarn2/Yarn2.kt

```diff
@@ -443,7 +443,7 @@ class Yarn2(
                 vcsFromPackage = vcsFromPackage.merge(details.vcsFromDownloadUrl)
             }
 
-            val id = Identifier("NPM", namespace, name, details.version)
+            val id = Identifier("NPM", namespace, name, version)
```

Comment:
Commit message: It's a bit odd to refer to an upcoming change that's not part of this PR. Also, just from looking at the diff it's not clear why `details.` can just be omitted, i.e. why `details.version` and `version` seem to refer to the same value.

Actually, as `version` is defined as

    val version = header.version.cleanVersionString()

and used as

    val details = packagesDetails["${header.rawName}@$version"]

I'd argue that `version` should *only* be used to create the key for the lookup in `packagesDetails`, and otherwise `details.version` should exclusively be used as the package version.

Reply:
> 2\. The `PackageJson.version` returned by that `yarn npm view` always is identical with the `version` requirested

Would you have a(n API) reference for that?

Anyway, would it make sense to postpone this commit to the PR that actually benefits from it? Right now I just don't see why replacing `details.version` with `version` is an improvement besides a few characters saved. To me, this is semantically mixing up "the version used for requesting details" with "the version returned as part of details" even if they may be the same.

Reply:
Ok, I've dropped it.